QUARKUS-5615.md

```diff
@@ -0,0 +1,30 @@
+## Links to the resources
+- JIRA ticket: https://issues.redhat.com/browse/QUARKUS-5615
+- Upstream documentation PR: https://github.com/quarkusio/quarkus/pull/46182
+- Follow-up to the original PR: https://github.com/quarkusio/quarkus/pull/46211
+- Serialisation PR: https://github.com/quarkusio/quarkus/pull/41063
+- Serialisation documentation PR: https://github.com/quarkusio/quarkus/pull/42289
+- Related guide: https://quarkus.io/guides/rest#reflection-free-jackson-serialization
+
+## Scope of the testing 
+This feature adds a new Quarkus property (`quarkus.rest.jackson.optimization.enable-reflection-free-serializers`) which should not affect behavior of Quarkus.
+If this property is activated for a project which uses Jackson for rest endpoints, but doesn't use advanced Jackson annotations (anything, besides @JsonProperty and @JsonIgnore), then Quarkus will use compile-time (de)serialisation, which is a bit faster, than the runtime-one.
+
+## Getting familiar with the feature
+- Read the guides:
+ * https://quarkus.io/guides/rest#reflection-free-jackson-serialization
+ * https://quarkus.io/blog/quarkus-metaprogramming/
+
+
+## Community status of the extension/feature:
+Affected extensions (REST and Jackson) are supported, but the feature itself is in Technology Preview.
+
+## Automated test development
+To get maximum coverage against breaking changes, add this property to every module of main TS, which uses Jackson. Add test method (one per module), which detects, that serialisation class was generated. Make sure, that this test doesn't affect compilation (avoid using reflection, try checking folder with generated classes instead).
```

Comment:
`To get maximum coverage against breaking changes, add this property to every module of main TS, which uses Jackson.` ... this means the default code path for serialization and deserialization won't be tested anymore.

Please think about different approach.

Note: For local experiment's to identify as many issues as possible, it a good strategy to add this property to every module of main TS that uses Jackson. But it's not ideal as a final thing to land in the TS.

Reply:
Suggested a different approach